lib/executor.js

```diff
@@ -1,60 +1,70 @@
-var log = require('./logger').create()
+'use strict'
 
-var Executor = function (capturedBrowsers, config, emitter) {
-  var self = this
-  var executionScheduled = false
-  var pendingCount = 0
-  var runningBrowsers
+const log = require('./logger').create()
 
-  var schedule = function () {
-    var nonReady = []
+class Executor {
+  constructor (capturedBrowsers, config, emitter) {
+    this.capturedBrowsers = capturedBrowsers
+    this.config = config
+    this.emitter = emitter
 
-    if (!capturedBrowsers.length) {
-      log.warn('No captured browser, open %s//%s:%s%s', config.protocol, config.hostname,
-        config.port, config.urlRoot)
+    this.executionScheduled = false
+    this.pendingCount = 0
+    this.runningBrowsers = null
+
+    // bind all the events
+    this.emitter.on('run_complete', () => this.onRunComplete())
```

Comment:
Use explicit event binding as in the previous PR. The end goal would be to remove `bindAllEvents` completely and use explicit subscriptions.